Modernize override

r-barnes opened the pull request:
Differential Revision: D29717320

malfet commented:
Please amend the PR by replacing `virtual ... override` with `... override` and prune irrelevant linter warnings

r-barnes committed:
Modernize override (#61744)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/61744

Test Plan: Sandcastle

Reviewed By: malfet

Differential Revision: D29717320

fbshipit-source-id: eb090adaef5fd70153bb8105be76ff0dee69ec77

malfet commented on torch/csrc/jit/codegen/cuda/ir_base_nodes.h:
```diff
@@ -223,8 +222,8 @@ class TORCH_CUDA_CU_API Val : public Statement {
 
-  // NOLINTNEXTLINE(cppcoreguidelines-explicit-virtual-functions,clang-diagnostic-inconsistent-missing-override,modernize-use-override)
-  virtual bool sameType(const Statement* other) {
+  // NOLINTNEXTLINE(cppcoreguidelines-explicit--functions,clang-diagnostic-inconsistent-missing-override)
```
`cppcoreguidelines-explicit--functions` looks like an incorrect nolint macro
```suggestion
```

malfet commented on test/cpp/tensorexpr/test_loopnest.cpp:
```diff
@@ -2245,7 +2245,7 @@ class LoopOrderHelper : public IRVisitor {
   }
 
-  // NOLINTNEXTLINE(cppcoreguidelines-explicit-virtual-functions,modernize-use-override)
+  // NOLINTNEXTLINE(cppcoreguidelines-explicit--functions,modernize-use-override)
```
`cppcoreguidelines-explicit--functions` looks like a typo (and perhaps nolint is no longer needed)
```suggestion
```

malfet commented on test/cpp/api/dataloader.cpp:
```diff
@@ -1051,11 +1051,11 @@ TEST(DataLoaderTest, MakeDataLoaderDefaultsAsExpected) {
     return {torch::ones(i), torch::ones(i)};
   }
-  // NOLINTNEXTLINE(cppcoreguidelines-explicit-virtual-functions,modernize-use-override)
+  // NOLINTNEXTLINE(cppcoreguidelines-explicit--functions,modernize-use-override)
   torch::optional<size_t> size() const noexcept {
```
```suggestion
  torch::optional<size_t> size() const noexcept override {
```
